src/GraphRequest.ts

```diff
@@ -501,19 +494,26 @@ export class GraphRequest {
 	 * @returns The same GraphRequest instance that is being called with
 	 */
 	public query(queryDictionaryOrString: string | KeyValuePairObjectStringNumber): GraphRequest {
-		const otherURLQueryParams = this.urlComponents.otherURLQueryParams;
+		let paramKey: string;
+		let paramValue: string | number;
 		if (typeof queryDictionaryOrString === "string") {
-			const querySplit = queryDictionaryOrString.split("=");
-			const queryKey = querySplit[0];
-			const queryValue = querySplit[1];
-			otherURLQueryParams[queryKey] = queryValue;
+			const indexOfFirstEquals = queryDictionaryOrString.indexOf("="); // The query key-value pair must be split on the first equals sign to avoid error while parsing nested query parameters
+			const qParams = [queryDictionaryOrString.substring(0, indexOfFirstEquals), queryDictionaryOrString.substring(indexOfFirstEquals + 1, queryDictionaryOrString.length)];
+			paramKey = qParams[0];
+			paramValue = qParams[1];
 		} else {
 			for (const key in queryDictionaryOrString) {
 				if (queryDictionaryOrString.hasOwnProperty(key)) {
-					otherURLQueryParams[key] = queryDictionaryOrString[key];
+					paramKey = key;
+					paramValue = queryDictionaryOrString[key];
 				}
 			}
 		}
+		if (oDataQueryNames.indexOf(paramKey) !== -1) {
+			this.urlComponents.oDataQueryParams[paramKey] = paramKey === "$expand" || paramKey === "$select" || paramKey === "$orderby" ? (this.urlComponents.oDataQueryParams[paramKey] ? this.urlComponents.oDataQueryParams[paramKey] + "," + paramValue : paramValue) : paramValue;
```

Comment:
oDataQueryParams were not being modified in the context of a `query` call, now they are. This seems to be an unexpected side effect.

I see that it was being modified in the `parsePath` function before, but this is a public function and you are changing the behavior not only for `parsePath` code path, but for all public callers.

Reply:
As explained in the previous comment, the idea is to maintain consistency. The reason to call the query function from parsePath function is to avoid duplicate comment.

Reply:
@ddyett @MIchaelMainer Would like to know your thoughts here please.

Reply:
Sounds like we need a private `parseQueryString`. 

Both `parsePath` and `query` functions need to parse a querystring and apply the odata and non-odata query parameters to `urlComponents` object. The `query` function had the same bug as `parsePath` where it split on  `=` without considering nested queries. 

The change to parsePath should be treated as public since it is always used to parse the path argument supplied to `client.api("")`.